### PR TITLE
ci: automatically remove `ok-to-test` and set it when queuing a PR

### PR DESCRIPTION
**Pull request by nixpanic**
The `ok-to-test` label currently needs to be removed and re-added when a PR is rebased for re-queuing.

It should be possible to automate this, by removing the `ok-to-test` label when a PR gets queued. It can automatically be added again when the PR has embarked the merge train.

Tested with the online Mergify Config Editor on PR #3505:

image

**Note:** we will need to keep an eye on CI usage when PRs are getting queued. Hopefully the "embarked the merge train" status does not happen when PRs do not need rebasing.

---

<details>
<summary>Show available bot commands</summary>

These commands are normally not required, but in case of issues, leave any of
the following bot commands in an otherwise empty comment in this PR:

- `/retest ci/centos/<job-name>`: retest the `<job-name>` after unrelated
  failure (please report the failure too!)
- `/retest all`: run this in case the CentOS CI failed to start/report any test
  progress or results

</details>

### CODE REVIEW

**Review by Madhu-1 — DISMISSED** [review]

**Review by yati1998 — DISMISSED** [review]

**Comment by nixpanic** [comment]
This needs manually merging, waiting for #3505 to get merged first.

**Comment by nixpanic** [comment]
added an extra condition for adding the `ok-to-check` label, only when on the `devel` or `release-*` branches.

File: .mergify.yml
Comment (Madhu-1) [inline .mergify.yml:49]: This will get added when we have the required approvals and only on one PR, which is in queue, right? Not on all the PRs? If it gets added to all the PRs in the queue again, its waste of resource

**Review by Madhu-1 — APPROVED** [review]

File: .mergify.yml
Comment (nixpanic) [reply .mergify.yml:49]: I am not sure if it gets added to all PRs, that is difficult to verify without testing it out first.

Comment (Madhu-1) [reply .mergify.yml:49]: Lets merge this and verify it out :dagger:

**Comment by nixpanic** [comment]
Let's try to get this in once #3506 is merged.

**Review by humblec — APPROVED** [review]

**Comment by humblec** [comment]
Merging this as the other one has been out of merge queue.

**Pull request merged**
